### PR TITLE
Web service for cluster node self-discovery, Dockerization, build scripting

### PR DESCRIPTION
**Pull request by heynickc**
Added a web service that new cluster nodes can use for self-discovery.  Installable as a Windows service via `Lighthouse.WhatsMyIP.exe install --localsystem --autostart`.  It will host a Web API endpoint listening on port `:9000/api/whatsmyip` that simply responds with the requester's URL with `scheme://host:port`.  It can be used by cluster nodes to set their `remote.dot-netty.tcp.public-hostname` when their actor systems get started.

Also added Docker support.  In the folder where the Dockerfile lives, you can run `docker build -t <dockerusername>/<imagename> .` to build an image for yourself locally.  It will install both the lighthouse and the whatsmyip service as windows services in the container.

Also build scripts for CI, for now only building and you can also specify `build.ps1 Docker` to build the docker image.  TODO will be to add tests and distributables like Chocolatey.

### CODE REVIEW

File: src/Lighthouse/App.config [hunk 1]
```diff
@@ -20,13 +20,13 @@
...
-							helios.tcp {
-								transport-class = "Akka.Remote.Transport.Helios.HeliosTcpTransport, Akka.Remote"
+							dot-netty.tcp {
+								transport-class = "Akka.Remote.Transport.DotNetty.TcpTransport, Akka.Remote"
 								applied-adapters = []
 								transport-protocol = tcp
 								#will be populated with a dynamic host-name at runtime if left uncommented
-								#public-hostname = "POPULATE STATIC IP HERE"
-								hostname = "0.0.0.0"
+								public-hostname = "172.22.144.2"
```
Comment (heynickc) [inline src/Lighthouse/App.config]: When using Docker to host Lighthouse, you have to call `docker run -di --ip=**172.22.144.2** --rm --name lighthouse nickc/lighthouse:latest` with the same static ip that is set here.  I've left this IP address since it works on the default Windows Container NAT.

**Comment by heynickc** [comment]
Until I extract the bootstrapper that will use the whatsmyip service, the WebCrawler demo has been updated to use the service to set its configs here:  https://github.com/heynickc/akkadotnet-code-samples/blob/webcrawler-dockerization/Cluster.WebCrawler/src/WebCrawler.Service/TrackerServiceFactory.cs

**Comment by heynickc** [comment]
Changed title to WIP while I pull in bootstrapper code from the client sample project I was using.  Also my upgrade to dot-netty transport didn't work as expected so reverting this.

**Comment by Aaronontheweb** [comment]
I would consolidate the `WhatIsMyIP.exe` into the same process as `Lighthouse.exe` itself - just expose the Web API functionality as part of the same service as the normal Lighthouse work.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# escape=`
+
+FROM microsoft/windowsservercore
+
+COPY .\src\Lighthouse\bin\Release C:\lighthouse\
+COPY .\src\Lighthouse.WhatsMyIP\bin\Release C:\lighthouse.whatsmyip\
+
+RUN ["C:\\lighthouse\\Lighthouse.exe","install","--localsystem","--autostart"]
```
Comment (heynickc) [inline Dockerfile]: It's relative for anyone building the project on their host machine, these are telling where the executable goes inside of the docker image once it's built.  So basically the paths are absolute *inside the image, but relative outside.

**Comment by heynickc** [comment]
Made some updates with consolidating the api into the same lighthouse process as lighthouse itself.  Also began adding the api to retrieve seed nodes, writing tests to cover edge cases with existing seed-nodes in the configuration vs without, etc.  Still WIP.

**Comment by Aaronontheweb** [comment]
@heynickc let me know when it's ready for next review

**Comment by heynickc** [comment]
@Aaronontheweb please review at your convenience.  TODO is CI and more tests.  Latest changes include new api endpoint for discovering *other seed nodes (other than the one hard coded into a non-seed node).   Renamed service to `LighthouseDiscoveryService` since it is multi-purpose.  More tests added for client configuration bootstrapping.

**Comment by Aaronontheweb** [comment]
@heynickc we have some conflicts that need solving here

**Comment by heynickc** [comment]
@Aaronontheweb I'm aware of the build failures.  Working to resolve.
